<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="ko" xml:lang="ko"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.294">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="William">
<meta name="description" content="각각 어떤 의미를 가지고 있는지 알아보자">

<title>윌리엄의 기술 블로그 - RPKM, FPKM, TPM이 무엇인가?</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<script src="../../site_libs/quarto-html/quarto.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "일치 없음",
    "search-matching-documents-text": "일치된 문서",
    "search-copy-link-title": "검색 링크 복사",
    "search-hide-matches-text": "추가 검색 결과 숨기기",
    "search-more-match-text": "추가 검색결과",
    "search-more-matches-text": "추가 검색결과",
    "search-clear-button-title": "제거",
    "search-detached-cancel-button-title": "취소",
    "search-submit-button-title": "검색"
  }
}</script>

<script type="text/javascript">

(function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
ga('create', 'UA-150213770-1', 'auto');

ga('send', {
  hitType: 'pageview',
  'anonymizeIp': true,
});
</script>


<link rel="stylesheet" href="../../styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../../index.html">
    <span class="navbar-title">윌리엄의 기술 블로그</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="검색"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="탐색 전환" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../about.html" rel="" target="">
 <span class="menu-text">홈</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../index.html" rel="" target="">
 <span class="menu-text">블로그</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../research.html" rel="" target="">
 <span class="menu-text">연구 아카이브</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../eng.html" rel="" target="">
 <span class="menu-text">Eng.Study</span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/williamjeong2" rel="" target=""><i class="bi bi-github" role="img">
</i> 
 <span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://www.linkedin.com/in/wjd5480/" rel="" target=""><i class="bi bi-linkedin" role="img">
</i> 
 <span class="menu-text"></span></a>
  </li>  
</ul>
            <div class="quarto-navbar-tools">
</div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">목차</h2>
   
  <ul>
  <li><a href="#rpkm" id="toc-rpkm" class="nav-link active" data-scroll-target="#rpkm">RPKM</a>
  <ul class="collapse">
  <li><a href="#정리해볼까요" id="toc-정리해볼까요" class="nav-link" data-scroll-target="#정리해볼까요">정리해볼까요?</a></li>
  </ul></li>
  <li><a href="#fpkm" id="toc-fpkm" class="nav-link" data-scroll-target="#fpkm">FPKM</a></li>
  <li><a href="#tpm" id="toc-tpm" class="nav-link" data-scroll-target="#tpm">TPM</a></li>
  <li><a href="#그래서-이게-뭘-의미하는데" id="toc-그래서-이게-뭘-의미하는데" class="nav-link" data-scroll-target="#그래서-이게-뭘-의미하는데">그래서 이게 뭘 의미하는데?</a>
  <ul class="collapse">
  <li><a href="#rpkm-1" id="toc-rpkm-1" class="nav-link" data-scroll-target="#rpkm-1">RPKM</a></li>
  <li><a href="#tpm-1" id="toc-tpm-1" class="nav-link" data-scroll-target="#tpm-1">TPM</a></li>
  <li><a href="#comparison-among-tpms" id="toc-comparison-among-tpms" class="nav-link" data-scroll-target="#comparison-among-tpms">Comparison among TPMs</a></li>
  <li><a href="#comparison-among-rpkms" id="toc-comparison-among-rpkms" class="nav-link" data-scroll-target="#comparison-among-rpkms">Comparison among RPKMs…?</a></li>
  <li><a href="#따라서" id="toc-따라서" class="nav-link" data-scroll-target="#따라서">따라서</a></li>
  <li><a href="#reference" id="toc-reference" class="nav-link" data-scroll-target="#reference">Reference</a></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">RPKM, FPKM, TPM이 무엇인가?</h1>
  <div class="quarto-categories">
    <div class="quarto-category">Bioinformatics</div>
  </div>
  </div>

<div>
  <div class="description">
    각각 어떤 의미를 가지고 있는지 알아보자
  </div>
</div>


<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">저자</div>
    <div class="quarto-title-meta-contents">
             <p>William </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">공개</div>
    <div class="quarto-title-meta-contents">
      <p class="date">2023년 3월 29일</p>
    </div>
  </div>
  
    
  </div>
  

</header>

<p>생물 정보학에서 굉장히 흔하게 접할 수 있는 용어 중 하나로 <strong>RPKM, FPKM, TPM</strong>이 있습니다.</p>
<p>RNA-seq의 발현 분석에서 normalized gene expression을 의미하는 것으로 초창기에는 RPKM이 많이 쓰였으나 FPKM을 거쳐 현재에는 TPM을 주로 사용하고 있습니다. (FPKM 또한 많이 사용중)</p>
<p>따라서 이 3가지 값을 계산하는 방법을 소개하려고 합니다. 아래 순서대로 이해하시면 됩니다.</p>
<p>먼저 A, B, C, D라는 4개의 gene과 3개의 replicate를 가지고 있는 테이블이 있습니다.<br>
gene 옆에 있는 kb(kilobase)는 gene의 길이입니다.</p>

<table>
<thead>
<tr>
<th>
Gene Name
</th>
<th>
Rep1 Counts
</th>
<th>
Rep2 Counts
</th>
<th>
Rep3 Counts
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
A (2kb)
</td>
<td>
10
</td>
<td>
12
</td>
<td>
30
</td>
</tr>
<tr>
<td>
B (4kb)
</td>
<td>
20
</td>
<td>
25
</td>
<td>
60
</td>
</tr>
<tr>
<td>
C (1kb)
</td>
<td>
5
</td>
<td>
8
</td>
<td>
15
</td>
</tr>
<tr>
<td>
D (10kb)
</td>
<td>
0
</td>
<td>
0
</td>
<td>
1
</td>
</tr>
</tbody>

</table>
<p>Rep3를 보면 gene에 상관없이 다른 replicate들보다 많은 reads를 가진 것을 알 수 있습니다. 이는 sequencing depth가 다른 replicate들보다 높다는 걸 의미합니다. 이제 이걸 normalize(정규화) 할 것입니다.</p>
<p>또한 gene B는 gene A에 2배에 해당하는 ’길이’를 가지고 있고, 이건 replicate에 상관없이 2배 많은 reads를 가지고 있다는 것을 설명하는 것일 수도 있습니다. 우리는 이것도 정규화해보려고 합니다.</p>
<section id="rpkm" class="level1">
<h1>RPKM</h1>
<p>RPKM은 single-end RNA-seq용으로 제작되었습니다. 먼저 사자성어 풀이하듯 각 알파벳의 의미를 먼저 살펴보겠습니다. <strong>R</strong>eads <strong>P</strong>er <strong>K</strong>ilobase per <strong>M</strong>illions mapped reads<br></p>
<p>그러면 이게 무슨 뜻이냐…하면 전체 유전자의 길이를 Kilobase(=1000 base pair)로 정규화했을 때 전체 reads 중에서 해당 gene에 mapping된 reads들을 말합니다. 그런데 이 수는 너무 적기 때문에 10<sup>6</sup>을 곱해주게 됩니다.<br></p>
<p>즉 FPKM은 total (1)reads에서 gene에 매핑된 리드들 나눈 값, (2) gene의 전체 길이를 1000으로 나눈 값, (3)10<sup>6</sup>을 곱해주면 됩니다.</p>
<p><img src="img/RPKM.png" class="quarto-discovered-preview-image img-fluid"><br></p>
<p>즉, gene 단위의 expression입니다.</p>
<p>이해가 잘 안되신다면 직접 계산해볼까요?<br></p>

<table>
<thead>
<tr>
<th>
Gene Name
</th>
<th>
Rep1 Counts
</th>
<th>
Rep2 Counts
</th>
<th>
Rep3 Counts
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
A (2kb)
</td>
<td>
10
</td>
<td>
12
</td>
<td>
30
</td>
</tr>
<tr>
<td>
B (4kb)
</td>
<td>
20
</td>
<td>
25
</td>
<td>
60
</td>
</tr>
<tr>
<td>
C (1kb)
</td>
<td>
5
</td>
<td>
8
</td>
<td>
15
</td>
</tr>
<tr>
<td>
D (10kb)
</td>
<td>
0
</td>
<td>
0
</td>
<td>
1
</td>
</tr>
</tbody>

</table>
<p>먼저, <strong>(1) read depth</strong>를 정규화합니다.</p>
<ul>
<li>각각의 replicate들에서 총 reads의 수를 계산합니다.</li>
</ul>

<table>
<thead>
<tr>
<th>
Gene Name
</th>
<th>
Rep1 Counts
</th>
<th>
Rep2 Counts
</th>
<th>
Rep3 Counts
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Total reads
</td>
<td>
35
</td>
<td>
45
</td>
<td>
106
</td>
</tr>
</tbody>

</table>
<ul>
<li>이 Total reads를 10<sup>6</sup>으로 나누어야 하는데 보기 쉽게 10으로만 나눠보겠습니다.</li>
</ul>

<table>
<thead>
<tr>
<th>
Gene Name
</th>
<th>
Rep1 Counts
</th>
<th>
Rep2 Counts
</th>
<th>
Rep3 Counts
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Tens of reads
</td>
<td>
3.5
</td>
<td>
4.5
</td>
<td>
10.6
</td>
</tr>
</tbody>

</table>
<p>따라서 이 값들은 각 replicate들에 대한 “per million” scaling factors입니다. 그러니까 우리는 각 gene의 read counts을 이 값들로 나누어야 합니다. 그러면 우리는 reads per million을 계산할 수 있는데 이를 <strong>RPM</strong>이라고 해보죠.</p>

<table>
<thead>
<tr>
<th>
Gene Name
</th>
<th>
Rep1 RPM
</th>
<th>
Rep2 RPM
</th>
<th>
Rep3 RPM
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
A (2kb)
</td>
<td>
2.86
</td>
<td>
2.67
</td>
<td>
2.83
</td>
</tr>
<tr>
<td>
B (4kb)
</td>
<td>
5.71
</td>
<td>
5.56
</td>
<td>
5.66
</td>
</tr>
<tr>
<td>
C (1kb)
</td>
<td>
1.43
</td>
<td>
1.78
</td>
<td>
1.42
</td>
</tr>
<tr>
<td>
D (10kb)
</td>
<td>
0
</td>
<td>
0
</td>
<td>
0.09
</td>
</tr>
</tbody>

</table>
<p><br><br></p>
<p>두 번째 단계는 <strong>(2) gene length로 정규화</strong>하는것입니다. 각 replicate의 counts를 gene length(<strong>K</strong>ilobase)로 나누면 됩니다. 쉽죠? 그러면 아래의 테이블이 나오게 됩니다.</p>

<table>
<thead>
<tr>
<th>
Gene Name
</th>
<th>
Rep1 RPKM
</th>
<th>
Rep2 RPKM
</th>
<th>
Rep3 RPKM
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
A (2kb)
</td>
<td>
1.43
</td>
<td>
1.33
</td>
<td>
1.42
</td>
</tr>
<tr>
<td>
B (4kb)
</td>
<td>
1.43
</td>
<td>
1.39
</td>
<td>
1.42
</td>
</tr>
<tr>
<td>
C (1kb)
</td>
<td>
1.43
</td>
<td>
1.78
</td>
<td>
1.42
</td>
</tr>
<tr>
<td>
D (10kb)
</td>
<td>
0
</td>
<td>
0
</td>
<td>
0.009
</td>
</tr>
</tbody>

</table>
<p>자 그러면 우리는 RPKM을 가지게 되었습니다.<br><br></p>
<section id="정리해볼까요" class="level3">
<h3 class="anchored" data-anchor-id="정리해볼까요">정리해볼까요?</h3>
<p>우리는 depth와 gene length을 정규화하지 않은 데이터를 가지고 있었습니다.</p>

<table>
<thead>
<tr>
<th>
Gene Name
</th>
<th>
Rep1 Counts
</th>
<th>
Rep2 Counts
</th>
<th>
Rep3 Counts
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
A (2kb)
</td>
<td>
10
</td>
<td>
12
</td>
<td>
30
</td>
</tr>
<tr>
<td>
B (4kb)
</td>
<td>
20
</td>
<td>
25
</td>
<td>
60
</td>
</tr>
<tr>
<td>
C (1kb)
</td>
<td>
5
</td>
<td>
8
</td>
<td>
15
</td>
</tr>
<tr>
<td>
D (10kb)
</td>
<td>
0
</td>
<td>
0
</td>
<td>
1
</td>
</tr>
</tbody>

</table>
<p>하지만 우리는 이제 각각의 sequencing depth와 각각의 gene length에 대해 정규화해서 각각의 replicate와 각각의 gene에 대한 RPKM 데이터를 가지게 되었습니다.</p>

<table>
<thead>
<tr>
<th>
Gene Name
</th>
<th>
Rep1 RPKM
</th>
<th>
Rep2 RPKM
</th>
<th>
Rep3 RPKM
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
A (2kb)
</td>
<td>
1.43
</td>
<td>
1.33
</td>
<td>
1.42
</td>
</tr>
<tr>
<td>
B (4kb)
</td>
<td>
1.43
</td>
<td>
1.39
</td>
<td>
1.42
</td>
</tr>
<tr>
<td>
C (1kb)
</td>
<td>
1.43
</td>
<td>
1.78
</td>
<td>
1.42
</td>
</tr>
<tr>
<td>
D (10kb)
</td>
<td>
0
</td>
<td>
0
</td>
<td>
0.009
</td>
</tr>
</tbody>

</table>
<p><br></p>
</section>
</section>
<section id="fpkm" class="level1">
<h1>FPKM</h1>
<p>RPKM과 유사한 이름을 가지고 있는 FPKM은 <strong>F</strong>ragment <strong>P</strong>er <strong>K</strong>ilobase of transcript per <strong>M</strong>illion mapped reads의 약자입니다. <strong>paired-end</strong>로 RNA-seq을 하면 하나의 fragment에서 2개의 read가 나옵니다. (<em>RPKM의 무조건 2배가 되는건 아닙니다</em>) 이는 transcript 관점의 expression으로 볼 수 있습니다.</p>
<p><img src="img/FPKM.png" class="img-fluid"></p>
<p>RPKM과 FPKM의 차이점은 FPKM은 두개의 reads가 하나의 fragment에 mapping된다는 점을 고려한다는 것입니다.</p>
<p><br></p>
</section>
<section id="tpm" class="level1">
<h1>TPM</h1>
<p>TPM은 Transcripts Per Million의 약자로 FPKM, RPKM과 유사한 개념이지만 transcript length의 분포까지 포함한 개념이라고 합니다.</p>
<p>TPM을 계산하는 방법 아래와 같습니다.</p>
<p>RPKM / Total RPKM * 10^6</p>
<p>흠.. 같이 봐볼까요?</p>

<table>
<thead>
<tr>
<th>
Gene Name
</th>
<th>
Rep1 Counts
</th>
<th>
Rep2 Counts
</th>
<th>
Rep3 Counts
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
A (2kb)
</td>
<td>
10
</td>
<td>
12
</td>
<td>
30
</td>
</tr>
<tr>
<td>
B (4kb)
</td>
<td>
20
</td>
<td>
25
</td>
<td>
60
</td>
</tr>
<tr>
<td>
C (1kb)
</td>
<td>
5
</td>
<td>
8
</td>
<td>
15
</td>
</tr>
<tr>
<td>
D (10kb)
</td>
<td>
0
</td>
<td>
0
</td>
<td>
1
</td>
</tr>
</tbody>

</table>
<p>먼저 (1) 각 replicate의 counts를 gene length로 나눕니다. 그러면 아래의 테이블이 나오겠죠? <strong>RPK</strong>는 <strong>R</strong>eads <strong>P</strong>er <strong>K</strong>ilobase라는걸 알 수 있습니다.</p>

<table>
<thead>
<tr>
<th>
Gene Name
</th>
<th>
Rep1 RPK
</th>
<th>
Rep2 RPK
</th>
<th>
Rep3 RPK
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
A (2kb)
</td>
<td>
5
</td>
<td>
6
</td>
<td>
15
</td>
</tr>
<tr>
<td>
B (4kb)
</td>
<td>
5
</td>
<td>
6.25
</td>
<td>
15
</td>
</tr>
<tr>
<td>
C (1kb)
</td>
<td>
5
</td>
<td>
8
</td>
<td>
15
</td>
</tr>
<tr>
<td>
D (10kb)
</td>
<td>
0
</td>
<td>
0
</td>
<td>
0.1
</td>
</tr>
</tbody>

</table>
<p>두 번째 단계로 (2) sequencing depth로 정규화하는 것입니다. 우리가 gene lengh을 정규화한 것들의 합(각 replicate read counts의 합)과 각 replicate의 RPK 수를 합하고 (원래는 10^6으로 나눠야 하지만) 10으로 나눈 Tens of RPK를 사용할 것입니다.</p>

<table>
<thead>
<tr>
<th>
Gene Name
</th>
<th>
Rep1 RPK
</th>
<th>
Rep2 RPK
</th>
<th>
Rep3 RPK
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Total RPK
</td>
<td>
15
</td>
<td>
20.25
</td>
<td>
45.1
</td>
</tr>
<tr>
<td>
Tens of RPK
</td>
<td>
1.5
</td>
<td>
2.025
</td>
<td>
4.51
</td>
</tr>
</tbody>

</table>
<p>각 replicate의 counts를 Tens of RPK로 나누게 되면 TPM이 완성됩니다.</p>

<table>
<thead>
<tr>
<th>
Gene Name
</th>
<th>
Rep1 TPM
</th>
<th>
Rep2 TPM
</th>
<th>
Rep3 TPM
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
A (2kb)
</td>
<td>
3.33
</td>
<td>
2.96
</td>
<td>
3.326
</td>
</tr>
<tr>
<td>
B (4kb)
</td>
<td>
3.33
</td>
<td>
3.09
</td>
<td>
3.326
</td>
</tr>
<tr>
<td>
C (1kb)
</td>
<td>
3.33
</td>
<td>
3.95
</td>
<td>
3.326
</td>
</tr>
<tr>
<td>
D (10kb)
</td>
<td>
0
</td>
<td>
0
</td>
<td>
0.02
</td>
</tr>
</tbody>

</table>
<p><br><br></p>
</section>
<section id="그래서-이게-뭘-의미하는데" class="level1">
<h1>그래서 이게 뭘 의미하는데?</h1>
<p>그렇죠 계산하는 방법만 알면 다가 아니죠. 이 둘(RPKM과 TPM)이 뭐가 다른지 알아야겠지요. RPKM과 TPM을 비교해봅시다.</p>
<section id="rpkm-1" class="level3">
<h3 class="anchored" data-anchor-id="rpkm-1">RPKM</h3>

<table>
<thead>
<tr>
<th>
Gene Name
</th>
<th>
Rep1 RPKM
</th>
<th>
Rep2 RPKM
</th>
<th>
Rep3 RPKM
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
A (2kb)
</td>
<td>
1.43
</td>
<td>
1.33
</td>
<td>
1.42
</td>
</tr>
<tr>
<td>
B (4kb)
</td>
<td>
1.43
</td>
<td>
1.39
</td>
<td>
1.42
</td>
</tr>
<tr>
<td>
C (1kb)
</td>
<td>
1.43
</td>
<td>
1.78
</td>
<td>
1.42
</td>
</tr>
<tr>
<td>
D (10kb)
</td>
<td>
0
</td>
<td>
0
</td>
<td>
0.009
</td>
</tr>
<tr>
<td>
Total
</td>
<td>
4.29
</td>
<td>
4.5
</td>
<td>
4.25
</td>
</tr>
</tbody>

</table>
</section>
<section id="tpm-1" class="level3">
<h3 class="anchored" data-anchor-id="tpm-1">TPM</h3>

<table>
<thead>
<tr>
<th>
Gene Name
</th>
<th>
Rep1 TPM
</th>
<th>
Rep2 TPM
</th>
<th>
Rep3 TPM
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
A (2kb)
</td>
<td>
3.33
</td>
<td>
2.96
</td>
<td>
3.326
</td>
</tr>
<tr>
<td>
B (4kb)
</td>
<td>
3.33
</td>
<td>
3.09
</td>
<td>
3.326
</td>
</tr>
<tr>
<td>
C (1kb)
</td>
<td>
3.33
</td>
<td>
3.95
</td>
<td>
3.326
</td>
</tr>
<tr>
<td>
D (10kb)
</td>
<td>
0
</td>
<td>
0
</td>
<td>
0.02
</td>
</tr>
<tr>
<td>
Total
</td>
<td>
10
</td>
<td>
10
</td>
<td>
10
</td>
</tr>
</tbody>

</table>
<p>위 결과들은 같은 데이터에서 나온 RPKM과 TPM 값입니다. 둘 다 gene length와 sequencing depth의 편향(bias)이 일치합니다. 하지만 각 column에 대한 정규화된 총 reads는 매우 다릅니다.</p>
<ol type="1">
<li>RPKM은 각 sample마다 다른 값을 얻었습니다.</li>
<li>TPM은 각 column마다 같은 값을 얻었습니다.</li>
</ol>
<p>왜 이 차이가 중요한지 봐봅시다. <br></p>
</section>
<section id="comparison-among-tpms" class="level3">
<h3 class="anchored" data-anchor-id="comparison-among-tpms">Comparison among TPMs</h3>
<p><img src="img/TPM_comp.png" class="img-fluid"></p>
</section>
<section id="comparison-among-rpkms" class="level3">
<h3 class="anchored" data-anchor-id="comparison-among-rpkms">Comparison among RPKMs…?</h3>
<p><img src="img/RPKM_comp.png" class="img-fluid"></p>
<p>감이 오시나요?</p>
<p>TPM은 모든 replicate에 대하여 총 count 수가 보정되었기 때문에 각 gene에 어떤 비율로 reads가 있는지 알 수 있습니다. 예를 들어 Rep1의 gene A는 33%라고 알 수 있습니다. 그리고 이를 통해 Rep2의 gene A와 비교하였을 때 Rep1의 gene A가 Rep2의 gene A보다 약간 크다고 알 수 있습니다. 이는 Rep1의 gene A에 맵핑된 양이 Rep3의 gene A에 맵핑된 양보다 미세하게 더 많다는 걸 알 수 있습니다.</p>
<p>하지만 RPKM은 각각의 replicate가 다른 total reads를 가지기 때문에 각각의 replicate의 gene을 비교하기 어렵습니다. 이 말은 각각의 gene의 reads 수를 비교하기 어렵다는 말이 됩니다.<br></p>
</section>
<section id="따라서" class="level3">
<h3 class="anchored" data-anchor-id="따라서">따라서</h3>
<p>sample들 안에서 맵핑되는 reads의 비율을 보다 명확하게 말해주는 <strong>TPM</strong>을 사용하기를 추천드립니다.</p>
<hr>
</section>
<section id="reference" class="level3">
<h3 class="anchored" data-anchor-id="reference">Reference</h3>
<ul>
<li><p>https://www.youtube.com/watch?v=TTUrtCY2k-w</p></li>
<li><p>https://www.rna-seqblog.com/rpkm-fpkm-and-tpm-clearly-explained/</p></li>
<li><p>http://www.incodom.kr/Expression_profiling</p></li>
</ul>
<hr>


</section>
</section>

<div id="quarto-appendix" class="default"><section class="quarto-appendix-contents"><h2 class="anchored quarto-appendix-heading">라이센스</h2><div id="quarto-reuse" class="quarto-appendix-contents"><div><a rel="license" href="https://creativecommons.org/licenses/by/4.0/deed.ko">https://creativecommons.org/licenses/by/4.0/deed.ko</a></div></div></section></div></main> <!-- /main -->
<div>
<button type="button" class="btn btn-primary btn-lg" onclick="window.open(&quot;https://toss.me/williamj&quot;)">💵Toss로 서포트하기</button>
<p class="text-muted">송금하실 때 메세지에 어떤 글을 보았는지 남겨주시면 게시글에 남겨드립니다. <br>기부해주신 돈은 커피를 마시는데 사용됩니다.</p>
</div>
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "복사완료!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "복사완료!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<script src="https://giscus.app/client.js" data-repo="williamjeong2/williamjeong2.github.io" data-repo-id="R_kgDOIMJu-w" data-category="Comments" data-category-id="DIC_kwDOIMJu-84CR6zt" data-mapping="pathname" data-reactions-enabled="1" data-emit-metadata="0" data-input-position="top" data-theme="light" data-lang="ko" crossorigin="anonymous" async="">
</script>
</div> <!-- /content -->



</body></html>